wallet/src/main/java/com/android/identity_credential/wallet/ui/prompt/passphrase/PassphrasePinScreen.kt

```diff
@@ -0,0 +1,364 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Text
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.identity.appsupport.ui.PassphraseEntryField
+import com.android.identity.securearea.PassphraseConstraints
+import com.android.identity_credential.wallet.R
+
+
+/**
+ * The buttons of KeyPad composable.
+ */
+val keyPadButtons = listOf(
+    // empty chars under 1 so it's top-aligned like digits 2-9
+    KeyPadButton(1, chars = " "),
+    KeyPadButton(2, "ABC"),
+    KeyPadButton(3, "DEF"),
+    KeyPadButton(4, "GHI"),
+    KeyPadButton(5, "JKL"),
+    KeyPadButton(6, "MNO"),
+    KeyPadButton(7, "PQRS"),
+    KeyPadButton(8, "TUV"),
+    KeyPadButton(9, "WXYZ"),
+    KeyPadButton(), //spacer
+    KeyPadButton(0)
+)
+
+/**
+ * The data of a KeyPad button defines the [number]/digit to show on top and the [chars] to show
+ * underneath the number/digit. Alternatively a KeyPad button can show if an image resource id
+ * [imageResId] is provided - this is paired with [buttonId] so it can be identified when the button
+ * is tapped along with the [description] of the image for accessibility.
+ */
+data class KeyPadButton(
+    // the numerical part of the keypad button (actual input number)
+    val number: Int? = null,
+    // the mnemonic part of the keypad button (letters under each number)
+    val chars: String? = null,
+    // show an image instead of a number and chars; provide buttonId below to find it in a list
+    val imageResId: Int? = null,
+    // used to identify the button when an imageResId is provided (and number and chars are null)
+    val buttonId: String? = null,
+    // for accessibility usually for image buttons
+    val description: String? = null,
+) {
+    /**
+     * An empty KeyPadButton renders a Spacer instead.
+     */
+    fun isEmpty() = number == null && chars == null && imageResId == null && description == null
+}
+
+/**
+ * Composes a full screen with PassphraseFieldEntry and custom numerical key pad to use for PINs.
```

Comment:
No need to enter `PassphraseFieldEntry`, that seems to be just an implementation detail.